core/controllers/editor.py

```diff
@@ -101,9 +101,53 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+class EditorLogoutHandler(base.BaseHandler):
+    """Handles logout from editor page."""
+
+    @acl_decorators.can_play_exploration
+    def is_valid_exploration_id(self, exp_id):
+        """This is to check if exploration id is valid."""
+        pass
+
+    @acl_decorators.open_access
+    def get(self):
+        """Checks if exploration is published and redirects accordingly."""
+
+        return_url = str(self.request.get('return_url'))
+        if return_url:
+            exploration_id = return_url.split('/')[-1]
+            self.is_valid_exploration_id(exploration_id)
+            exp_summary = exp_services.get_exploration_summary_by_id(
+                exploration_id)
+
+            if exp_summary.status == feconf.ACTIVITY_STATUS_PRIVATE:
```

Comment:
Minimize the amount of code that's being branched on.
```
url_to_redirect_to = (
    ... if exp_summary.is_private() else ...)
self.redirect(url_to_redirect_to)
```

Reply:
i have cleaned to code.